resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -137,11 +153,18 @@ public DnsCacheEntry cache(String hostname, DnsRecord[] additionals,
         checkNotNull(hostname, "hostname");
         checkNotNull(address, "address");
         checkNotNull(loop, "loop");
-        DefaultDnsCacheEntry e = new DefaultDnsCacheEntry(hostname, address);
+
+        int ttl = Math.max(minTtl, (int) Math.min(maxTtl, originalTtl));
+        long ttlInMillis = TimeUnit.SECONDS.toMillis(ttl);
+        long softExpiryTime = System.currentTimeMillis() + ttlInMillis;
```

Comment:
I think it would be better to use `System.nanoTime()` to not be effected by time changes